### ISSUE TITLE
Data Chip Crafting issue.

### ISSUE THREAD

**Issue by wiloxe** [LambdaInnovation/AcademyCraft#199]
So the title says most of the part however I'll attempt to provide more information
So first i can't check the recipe in NEI. When pressing R on Data Chip nothing happens.
Second, I found an crafting file telling how to craft the chip but would not appear in the finished crafting box.
Also same thing goes for Empty unit. Forge version: 1541 [Edited, forgot to put it in...]

Here's the log you asked for: http://bit.ly/1NMyrm1
NOTE: I could not use pastebin. Why? Well after pasting in the text the website keep's freezing for me.

**Comment by WeAthFoLD** [comment]
Thanks, I'll look into the issue.

PS.. Maybe the site's mystery bug? I don't know xD

**Comment by 1skandar** [comment]
Yup, same here. Several missing recipes despite using the latest Forge.

**Comment by 1skandar** [comment]
Ok, full list of missing recipes:
Data Chip
Normal Ability Developer 
Wireless Imag Matrix
Wind Generator Pillar
Machine Frame
Brainwave Analyzer
Resonator
App-Skill Tree

and, yes, this is the latest Forge.

**Comment by WeAthFoLD** [comment]
Actually I can't reproduce it... I'll try more on several version of JRE and check if its potential mod conflict.
@ShieLian Do me a favor?

**Comment by WeAthFoLD** [comment]
Also can you guys please provide your mod list?

**Comment by 1skandar** [comment]
http://forum.feed-the-beast.com/threads/1-7-10-modular-mayhem.120584/

**Comment by 1skandar** [comment]
Ah, wait, wrong mod list. Well, pretty much that mod list plus AcademyCraft.

**Comment by WeAthFoLD** [comment]
Cheers, I've nailed down the issue xD

Turns out Forge's ``OreDictionary.doesOreNameExist(id)`` returns ``true`` even when the id exists but the list is empty. Some mod thouches that id at initialization stage, but never stored any actual stack into the id. Then the ``RecipeRegistry`` of AC just turned to use that id directly rather than find the item associated with the name.

Adding a ``list.size() == 0`` check effectively solves this.

Also will change all recipes containing non-standard oreDict names into oreDict names.
